packages/flutter_tools/lib/src/build_system/targets/macos.dart

```diff
@@ -530,6 +549,16 @@ class ProfileMacOSBundleFlutterAssets extends MacOSBundleFlutterAssets {
     CompileMacOSFramework(),
     ProfileUnpackMacOS(),
   ];
+
+  @override
+  List<Source> get inputs => super.inputs + const <Source>[
```

Comment:
This looks right, but can you confirm that in your testing this doesn't run on subsequent compilations?